### DISCUSSION TITLE
mg_mgr_poll() does not block on Win32 with no existing connections

### DISCUSSION THREAD

**Discussion by pmeskoni** [cesanta/mongoose#1537] [Ask questions, get answers] [answered]
We call mg_mgr_poll() in a loop on the windows platform, regardless of whether we have any open connections. We were expecting that the timeout passed as an argument to mg_mgr_poll() would throttle the loop. It does, as long as there are active connections. We needed to do this to provide consistent throttling:

    if (m_mgr->conns != NULL) {
    	mg_mgr_poll(m_mgr, m_mgPollTime);
    }
    else {
    	sleep( m_mgPollTime );
    }

This is not the behavior on Linux or other platforms. It seems the windows socket library is responsible for this behavior, but it could be mitigated by putting logic like this in Mongoose for windows only.

Would it make sense to include this logic in Mongoose for the windows platform to make the behavior consistent?

Thanks,
PM

**Comment by cpq** [comment, marked as answer]
@pmeskoni please use the latest HEAD, it should already be there.
